riscv/src/compiler.rs

```diff
@@ -635,7 +635,78 @@ fn preamble(degree: u64, coprocessors: &CoProcessors, with_bootloader: bool) ->
 "#
 }
 
-fn memory() -> String {
+fn memory(with_bootloader: bool) -> String {
+    // There are subtle differences between the memory machines with and without continuations:
+    // - There is an extra `mstore_bootloader` instruction. For the most part, it behaves just
+    //   like `mstore`.
+    // - When `m_change` is true, the `m_is_bootloader_write` has to be true in the next row.
+    // - The `(1 - m_is_write') * m_change * m_value' = 0` constraint is removed, as we no longer can
+    //   have a read as the first operation on a new address.
```

Comment:
Is this for any write or just for mstore_bootloader?

Reply:
The idea of the `(1 - m_is_write') * m_change * m_value' = 0` constraint is that when the address changes, either the first operation is a write (which can set any value), or the value is 0. But with the bootloader, we already constrain that it *cannot be anything other than a bootloader write*, so this rule becomes obsolete.